### PR TITLE
Update to .NET core 3.0

### PR DESCRIPTION
**Pull request by peppy**

### CODE REVIEW

File: osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs [hunk 1]
```diff
@@ -113,19 +113,14 @@ public void TestReadToEndNoPeeks()
...
 
             using (var stream = new MemoryStream(Encoding.UTF8.GetBytes(contents)))
             using (var bufferedReader = new LineBufferedReader(stream))
             {
                 Assert.AreEqual("this line is gone", bufferedReader.ReadLine());
                 Assert.AreEqual("this one shouldn't be", bufferedReader.PeekLine());
-                const string ending = @"this one shouldn't be
-these ones
-definitely not";
+                const string ending = "this one shouldn't be\nthese ones\ndefinitely not";
```
Comment (smoogipoo) [inline osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs]: Needs investigation as to why `ReadToEnd()` is providing a different output to the input above.

Comment (peppy) [reply osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs]: @bdach if you beat me to it, wanna take a look into this? not sure if it's a regression in .NET core 3.0 or an expected change (seems that `\r\n` are being replaced with `\n` on macOS, at least).

Comment (bdach) [reply osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs]: Sure, I'll have a look at it as the first thing today.

Comment (peppy) [reply osu.Game.Tests/Beatmaps/IO/LineBufferedReaderTest.cs]: Looks like this is also reproducing on master, so may have just never been run/tested on a non-windows host.

**Review by smoogipoo — APPROVED** [review]

**Comment by peppy** [comment]
Going to merge this one even with the random test failure (will fix tomorrow). I'd like to test the deploy process and see how we fare with iOS/android crashes.

**Pull request merged**